airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -96,6 +106,67 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             return False, str(e)
         return True, None
 
+    def _create_concurrent_stream(self, base_stream: HttpStream, concurrency_level: int = _MAX_CONCURRENCY) -> Stream:
+        http_requester = HttpRequester(
+            url_base=base_stream.url_base,
+            request_options_provider=InterpolatedRequestOptionsProvider(
+                request_headers={**{"Stripe-Version": "", "Stripe-Account": ""}, **base_stream.authenticator.get_auth_header()},
+                parameters={},
+            ),
+            path=base_stream.path(),
+            name=base_stream.name,
+            config={},
+            parameters={},
+            message_repository=self.message_repository,
+        )
+
+        paginator = DefaultPaginator(
+            CursorPaginationStrategy(
+                cursor_value="{{ last_records[-1]['id'] if last_records else None }}",
```

Comment:
@brianjlai would love your thoughts on this since we'll be doing some more PoCs...

On one hand I love that we can reuse the logic from the declarative framework and I'm a big fan of this new shape, but I'm very worried that using jinja macros from Python connectors will hinder maintenance. 

It might be worth finding a way to abstract the interpolated strings so we can use the components with real Python functions

Reply:
we synced offline and agreed it'll become spaghetti-esque to depend on low-code abstractions.

I'm pretty sure we can extract the string interpolation out of the pagination strategy object and a concrete stripe strategy that might look like
```
class StripePaginationStrategy(CursorPaginationStrategy):
    def stop(self, response, headers, last_records) -> bool:
        return "has_more" not in response

    def get_cursor_value(self, response, headers, last_records) -> str:
        if response["has_more"] and response.get("data", []):
            last_object_id = response["data"][-1]["id"]
            return last_object_id
        return None
```

I did a quick sketch on [this commit](https://github.com/airbytehq/airbyte/commit/8221fb5c061a8fe8b905038ab106d06508a4884d), but the main thing to verify is we can slowly update the classes from the declarative package without a big-bang refactor